File: Algorithms/1249_Minimum_Remove_to_Make_Valid_Parentheses/Solution.java
import java.util.Stack;

public class Solution {
    public String minRemoveToMakeValid(String s) {
        String[] cs = s.split("");
        Stack<Integer> stk = new Stack<>();
        for (int i = 0; i < cs.length; i++) {
            if ("(".equals(cs[i])) {
                stk.push(i);
            } else if (")".equals(cs[i])) {
                if (stk.isEmpty()) {
                    cs[i] = "";
                } else {
                    stk.pop();
                }
            }
        }
        while (!stk.isEmpty()) {
            cs[stk.pop()] = "";
        }
        return String.join("", cs);
    }
}

File: Algorithms/1249_Minimum_Remove_to_Make_Valid_Parentheses/solution.py
from typing import *


class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        cs: List[str] = [c for c in s]
        stk: List[int] = []

        for i in range(0, len(cs)):
            if cs[i] == '(':
                stk.append(i)
            elif cs[i] == ')':
                if stk:
                    stk.pop()
                else:
                    cs[i] = ''

        for i in range(0, len(stk)):
            cs[stk[i]] = ''

        return ''.join(cs)
